src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -857,6 +857,12 @@ void VM_Version::get_processor_features() {
   }
 #endif
 
+  // Check if processor has Intel Ecore
+  if (FLAG_IS_DEFAULT(DoEcoreOpt) && is_intel() && cpu_family() == 6 &&
+    (_model == 0x97 || _model == 0xAC || _model == 0xAF)) {
+    FLAG_SET_DEFAULT(DoEcoreOpt, true);
+  }
```

Comment:
And what should happen if the flag is set true by the user and there is no Ecore? What affect will that have? Should it be allowed?

Reply:
From ISA point of view, they are the same, so if user flips flag on purpose, code will still be correct. Its also helpful to test Ecore optimized code on a Pcore (I have some more patches coming in under this option soon)